Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -0,0 +1,173 @@
+  have h₂ : (-s).re + r ≤ 0 := by
+    rw [neg_re, neg_add_nonpos_iff]
+    exact hs.le
+  have h₃ : ∀ t ∈ Set.Ici (1 : ℝ), DifferentiableAt ℝ (fun x : ℝ ↦ ‖(x : ℂ) ^ (-s)‖) t := by
+    intro t ht
+    have ht' : t ≠ 0 := (zero_lt_one.trans_le ht).ne'
+    exact (differentiableAt_id.ofReal_cpow_const ht' h₁).norm ℝ <|
+      (cpow_ne_zero_iff_of_exponent_ne_zero h₁).mpr <| ofReal_ne_zero.mpr ht'
+  have h₄ : (deriv fun t : ℝ ↦ ‖(t : ℂ) ^ (- s)‖) =ᶠ[atTop] fun t ↦ - s.re * t ^ (- (s.re +1)) := by
+    filter_upwards [eventually_gt_atTop 0] with t ht
+    rw [deriv_norm_ofReal_cpow _ ht, neg_re, neg_add']
+  change Summable (fun n ↦ LSeries.term f s n)
+  simp_rw [LSeries.term_def₀ hf, mul_comm (f _)]
+  refine summable_mul_of_bigO_atTop' (f := fun t ↦ (t : ℂ) ^ (-s))
+    (g := fun t ↦ t ^ (- (s.re + 1) + r)) _ h₃ ?_ ?_ ?_ ?_
+  · refine integrableOn_Ici_iff_integrableOn_Ioi.mpr
+      (integrableOn_Ioi_deriv_norm_ofReal_cpow zero_lt_one ?_)
+    exact neg_re _ ▸ neg_nonpos.mpr  <| hr.trans hs.le
+  · refine (IsBigO.mul_atTop_rpow_natCast_of_isBigO_rpow ((- s).re) r 0 ?_ hO h₂).congr_right
+      (by simp)
+    exact (norm_ofReal_cpow_eventually_eq_atTop _).isBigO.natCast_atTop
+  · refine IsBigO.mul_atTop_rpow_of_isBigO_rpow (- (s.re + 1)) r _ ?_ ?_ le_rfl
+    · exact (EventuallyEq.isBigO h₄).of_const_mul_right
+    · exact (hO.comp_tendsto tendsto_nat_floor_atTop).trans <|
+        isEquivalent_nat_floor.isBigO.rpow hr (eventually_ge_atTop 0)
+  · exact integrableAtFilter_rpow_atTop_iff.mpr
+      (by rwa [neg_add_lt_iff_lt_add, add_neg_cancel_right])
+
+/-- If the partial sums `∑ k ∈ Icc 1 n, ‖f k‖` are `O(n ^ r)` for some real `0 ≤ r`, then the
+L-series `LSeries f` converges at `s : ℂ` for all `s` such that `r < s.re`. -/
+theorem LSeriesSummable_of_sum_norm_bigO
+    (hO : (fun n ↦ ∑ k ∈ Icc 1 n, ‖f k‖) =O[atTop] fun n ↦ (n : ℝ) ^ r)
+    (hr : 0 ≤ r) (hs : r < s.re) :
+    LSeriesSummable f s := by
+  have h₁ : (fun n ↦ if n = 0 then 0 else f n) =ᶠ[atTop] f := by
+    filter_upwards [eventually_ne_atTop 0] with n hn using by simp_rw [if_neg hn]
+  refine LSeriesSummable.congr' _ h₁ ?_
+  refine LSeriesSummable_of_sum_norm_bigO_aux (by rw [if_pos rfl]) ?_ hr hs
+  refine hO.congr' (Eventually.of_forall fun _ ↦ Finset.sum_congr rfl fun _ h ↦ ?_) EventuallyEq.rfl
+  rw [if_neg (zero_lt_one.trans_le (mem_Icc.mp h).1).ne']
+
+/-- If `f` takes nonnegative real values and the partial sums `∑ k ∈ Icc 1 n, f k` are `O(n ^ r)`
+for some real `0 ≤ r`, then the L-series `LSeries f` converges at `s : ℂ` for all `s`
+such that `r < s.re`. -/
+theorem LSeriesSummable_of_sum_norm_bigO_and_nonneg
+    {f : ℕ → ℝ} (hO : (fun n ↦ ∑ k ∈ Icc 1 n, f k) =O[atTop] fun n ↦ (n : ℝ) ^ r)
+    (hf : ∀ n, 0 ≤ f n) (hr : 0 ≤ r) (hs : r < s.re) :
+    LSeriesSummable (fun n ↦ f n) s := by
+  refine LSeriesSummable_of_sum_norm_bigO ?_ hr hs
+  simp_rw [norm_real, Real.norm_of_nonneg (hf _)]
+  exact hO
```

Comment:
```suggestion
    LSeriesSummable (fun n ↦ f n) s := 
  LSeriesSummable_of_sum_norm_bigO (by simpa [_root_.abs_of_nonneg (hf _)]) hr hs
```
(`Complex.abs_of_nonneg` should probably be `protected` to avoid this horrible `_root_` workaround, but that's a matter for another PR)